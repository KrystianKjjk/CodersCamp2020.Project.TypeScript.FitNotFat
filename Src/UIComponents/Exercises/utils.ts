import { ExercisesDetails } from "../../../Models/ExercisesDetails.model";
import { User } from "../../../Models/User.model";
import { ExerciseItemFromAPI } from "../../APIConnection/Exercises";

// data transforms
function prepareDataForTable(data: ExercisesDetails): string[] {
    return [
        data.name, 
        `${data.met}`, 
        `${data.duration}`, 
        `${data.calories} kcal`
    ];
}

function prepareAPIData(itemFromAPI: ExerciseItemFromAPI): ExercisesDetails {
    return {
        name: itemFromAPI.name,
        met: itemFromAPI.met,
        duration: itemFromAPI.duration_min,
        calories: itemFromAPI.nf_calories
    };
}

export { prepareDataForTable, prepareAPIData };