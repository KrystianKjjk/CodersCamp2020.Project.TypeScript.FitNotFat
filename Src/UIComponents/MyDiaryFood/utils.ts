import { FoodDetails } from "../../../Models/FoodDetails.model";
import { User } from "../../../Models/User.model";
import { FoodItemFromAPI } from "../../APIConnection/Food";

function sameDay(d1: Date | string, d2: Date | string) {
    const date1 = d1 instanceof Date ? d1 : new Date(d1);
    const date2 = d2 instanceof Date ? d2 : new Date(d2);
    
    return date1.getFullYear() === date2.getFullYear() 
        && date1.getMonth() === date2.getMonth() 
        && date1.getDate() === date2.getDate();
}

function isUserAuthorizedToUseApi(userData: User): boolean {
    return !!(userData && userData.detailsAPI && userData.detailsAPI.id && userData.detailsAPI.key);
}

function getApiCredentialsForUser(userData: User): {appKey: string, appId: string} | null {
    if (!isUserAuthorizedToUseApi(userData)) return null;

    return {
        appKey: userData.detailsAPI.key,
        appId: userData.detailsAPI.id,
    };
}

// data transforms
function prepareDataForTable(data: FoodDetails): string[] {
    return [
        data.name, 
        `${data.amount}`, 
        data.unit, 
        `${data.calories} kcal`
    ];
}

function prepareAPIData(itemFromAPI: FoodItemFromAPI): FoodDetails {
    return {
        name: itemFromAPI.food_name,
        amount: itemFromAPI.serving_qty,
        unit: itemFromAPI.serving_unit,
        calories: itemFromAPI.nf_calories
    };
}

function generateUniqueClassName(mealName: string, className: string) {
    return `${className}-${mealName}`;
}

export { sameDay, isUserAuthorizedToUseApi, getApiCredentialsForUser, prepareDataForTable, prepareAPIData, generateUniqueClassName };