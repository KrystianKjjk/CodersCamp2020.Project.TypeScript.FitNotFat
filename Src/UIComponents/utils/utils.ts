import { placeholder } from "@babel/types";
import { User } from '../../../Models/User.model'
import { KEY_LOGGED_USER } from "../../../Constants/consts";

export function createElement(tagName: string, classNames?: string[] | string, innerHTML?: string, dataComponent?: string): HTMLElement {
    const element = document.createElement(tagName);
    if(classNames){
        if(classNames instanceof(Array)) element.classList.add(...classNames);
        else element.classList.add(classNames);
    }
    if(innerHTML) element.innerHTML = innerHTML;
    if(dataComponent) element.setAttribute('data-component', dataComponent);
    return element;
}

export function createInput(className: string, type: string, placeholder?: string): HTMLInputElement {
    const element = createElement('input', className) as HTMLInputElement;
    element.type = type;
    if(placeholder) element.placeholder = placeholder;
    return element;
}

export function createTextInput(placeholderValue: string, className: string):HTMLInputElement {
    const textInput = createInput(className, 'text', placeholderValue) as HTMLInputElement;
    textInput.required = true;
    return textInput;
}

export function createNumberInput(
    placeholderValue: string,
    className: string,
  ): HTMLInputElement {
    const numberInput = createInput(className, 'number', placeholderValue);
    numberInput.setAttribute('min', '0');
    numberInput.required = true;
    return numberInput;
}

export function createRadioInput(className: string, name: string, inputValue: string): HTMLInputElement {
    const radioInput = createElement('input', className) as HTMLInputElement;
    radioInput.setAttribute('type', 'radio');
    radioInput.setAttribute('name', name);
    radioInput.setAttribute('value', inputValue);
    radioInput.setAttribute('id', inputValue);
    radioInput.required = true;
    return radioInput;
  }

  //to calculate max calories per day
export function calculateCalories(gender: User["gender"], weight: number, height: number, age: number, activityLevel: User["activityLevel"]):number{
    let calories: number;

    if (gender === "Male") {
        calories = 66.47 + 13.7*weight + 5.0*height - 6.76*age; 

        switch (activityLevel){
            case 'Low':
                calories*=1.3;
                break;
            case 'Regular':
                calories*=1.6;
                break;
            case 'Active':
                calories*=1.7;
                break;
            case 'Very Active':
                calories*=2.1;
                break;
        }  
    } else {
        calories = 665.1 + 9.567*weight + 1.85*height - 4.68*age; 

        switch (activityLevel){
            case 'Low':
                calories*=1.3;
                break;
            case 'Regular':
                calories*=1.5;
                break;
            case 'Active':
                calories*=1.6;
                break;
            case 'Very Active':
                calories*=1.9;
                break;
        }  
    }   

    return calories;
}

export function getAge(date: Date) {     
    return ((new Date(Date.now() - date.getTime()).getFullYear()) - 1970);
}
export function getLoggedInUser(): string{
    return localStorage.getItem(KEY_LOGGED_USER) || '';
}

export function setLoggedInUser(loggedUser: string): string{
    const loggedInUser = getLoggedInUser();
    if(loggedInUser) return loggedInUser;
    localStorage.setItem(KEY_LOGGED_USER, loggedUser);
    return loggedUser;
}

export function clearLoggedInUser(): boolean{
    localStorage.setItem(KEY_LOGGED_USER, '');
    return true;
}
