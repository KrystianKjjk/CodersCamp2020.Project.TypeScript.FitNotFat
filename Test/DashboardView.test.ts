import dashboardView from '../Src/UIComponents/DashboardView/DashboardView';
import {createElement} from '../Src/UIComponents/utils/utils';

describe('test dashboardView ', () => {
    test('render correctly with second header', () => {
        const content = createElement('div', [], 'Content') as HTMLDivElement;
        const view = dashboardView('View', content, 'Second header!');
        expect(view).toMatchSnapshot('dashboardViewWith2ndHeader');
    })
    test('render correctly without second header', () => {
        const content = createElement('div', [], 'Content') as HTMLDivElement;
        const view = dashboardView('View', content);
        expect(view).toMatchSnapshot('dashboardViewWithout2ndHeader');
    })
})