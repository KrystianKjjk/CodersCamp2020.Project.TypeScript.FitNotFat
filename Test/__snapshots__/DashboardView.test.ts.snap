// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test dashboardView  render correctly with second header: dashboardViewWith2ndHeader 1`] = `
<div
  class="dashboard-view"
>
  <h1>
    View
  </h1>
  <h2>
    Second header!
  </h2>
  <div>
    Content
  </div>
</div>
`;

exports[`test dashboardView  render correctly without second header: dashboardViewWithout2ndHeader 1`] = `
<div
  class="dashboard-view"
>
  <h1>
    View
  </h1>
  <div>
    Content
  </div>
</div>
`;
